apps/web/components/eventtype/EventAdvancedTab.tsx

```diff
@@ -418,6 +429,103 @@ export const EventAdvancedTab = ({ eventType, team }: Pick<EventTypeSetupProps,
           </div>
         )}
       </SettingsToggle>
+      <SettingsToggle
+        labelClassName="text-sm"
+        toggleSwitchAtTheEnd={true}
+        switchContainerClassName={classNames(
+          "border-subtle rounded-lg border py-6 px-4 sm:px-6",
+          singleUseLinksVisible && "rounded-b-none"
+        )}
+        childrenClassName="lg:ml-0"
+        data-testid="singleUseLinksCheck"
+        title={t("single_use_links_title")}
+        description={t("single_use_links_description", { appName: APP_NAME })}
+        checked={singleUseLinksVisible}
+        onCheckedChange={(e) => {
+          if (!e) {
+            // Note - This will reset the single-use-links to empty, when it is toggled.
+            // If we need to persist the single-use-links when toggle is turned on and off, then we need to store the links in a state.
+            formMethods.setValue("singleUseLinks", [], { shouldDirty: true });
+          }
+          setSingleUseLinksVisible(e);
+        }}>
+        <div className="border-subtle rounded-b-lg border border-t-0 p-6">
+          <Controller
+            name="singleUseLinks"
+            render={({ field: { value, onChange } }) => {
+              const addSingleUseLink = () => {
+                let seed = String(formMethods.getValues("users")[0]?.id ?? team?.id);
+                // Append the last generated link to the seed for more randomized link generation.
+                if (value.length > 0) {
+                  seed = `${seed}:${value[value.length - 1]}`;
+                } else {
+                  // If this is the first one then just append a constant string to the seed.
+                  seed = `${seed}:single_use`;
+                }
```

Comment:
Is all this really necessary?

Reply:
Actually I wrote it to give it to make the seed more unique, though it was appended/prepended with the current time.

Thinking about it now.. I did over-engineer here.. 😛 Removed it!